drivers/hv/mshv_vtl_main.c

```diff
@@ -829,35 +829,22 @@ void mshv_tdx_request_cache_flush(bool wbnoinvd)
 	__tdx_hypercall(&args);
 }
 
-#define TDCALL_ASM	".byte 0x66,0x0f,0x01,0xcc"
-
-/* TODO TDX: Confirm noinline produces the right asm for saving register state */
-noinline void mshv_vtl_return_tdx(void)
+void mshv_vtl_return_tdx(void)
 {
 	struct tdx_tdg_vp_enter_exit_info *tdx_exit_info;
 	struct tdx_vp_state *tdx_vp_state;
 	struct mshv_vtl_run *vtl_run;
 	struct mshv_vtl_per_cpu *per_cpu;
-
-	register void *__sp asm("rsp");
-	register u64 r8 asm("r8");
-	register u64 r9 asm("r9");
-	register u64 r10 asm("r10");
-	register u64 r11 asm("r11");
-	register u64 r12 asm("r12");
-	register u64 r13 asm("r13");
-	register u64 r14 asm("r14");
-	register u64 r15 asm("r15");
+	struct tdx_module_args args;
```

Comment:
I'm not sure which is better to use struct tdx_module_args or tdx_tdg_vp_enter_exit_info.
which is consistent?

Reply:
We don't want to use tdx_tdg_vp_enter_exit_info to pass rcx and rdx arguments so created tdx_module_args. We ca pass tdx_tdg_vp_enter_exit_info  as argument to receive the exit info values directly.  

The input parameters rcx and rdx, they can be either passed directly as 2 parameters or inside tdx_module_args as following. Any preference?
__tdg_vp_enter(u64 rcx, u64 rdx, tdx_tdg_vp_enter_exit_info  *tdx_exit_info);

or
__tdg_vp_enter(tdx_module_args *args, tdx_tdg_vp_enter_exit_info  *tdx_exit_info);